src/commander/mod.rs

```diff
@@ -170,9 +175,43 @@ impl Commander {
         Ok(())
     }
 
+    #[instrument(level = "trace", skip(self))]
     pub fn init(&self) -> Result<()> {
-        self.execute_void_jj_command(vec!["status"])
-            .context("Failed getting initial status")
+        // Ask jj about its version
+        let (color, quiet) = (false, false);
+        let found_version = self
+            .execute_jj_command(vec!["version"], color, quiet)
+            .context("Run jj version")?;
+
+        // Extract version number
+        if found_version[0..3] != *"jj " {
+            trace!("jj version output \"{}\"", found_version);
+            bail!("Jujutsu version string was not recognized");
```

Comment:
This function should not throw errors if it fails to recognize version since this is used for diagnostic, not for restriction. For example - old jj version can not work for bookmarks, but everything else works fine
```suggestion
            return Ok(());
```